common/util/utils.go

```diff
@@ -82,17 +82,9 @@ func ToChaincodeArgs(args ...string) [][]byte {
 // ConcatenateBytes is useful for combining multiple arrays of bytes, especially for
 // signatures or digests over multiple fields
 func ConcatenateBytes(data ...[]byte) []byte {
-	finalLength := 0
+	var result []byte
 	for _, slice := range data {
-		finalLength += len(slice)
-	}
-	result := make([]byte, finalLength)
-	last := 0
-	for _, slice := range data {
```

Comment:
if you want to simplify this implementation then you can use the `copy()` built-in directive instead of this for loop

Reply:
Use copy() in the new version